src/components/KCodeBlock/KCodeBlock.vue

```diff
@@ -5,207 +5,150 @@
+              data-testid="clear-query-button"
+              role="button"
+              tabindex="0"
+              title="Clear query"
+              @click="clearQuery"
+              @keydown.enter="clearQuery"
+              @keydown.space="clearQuery"
+            />
+            <SearchIcon
+              v-else
+              class="code-block-search-icon"
+            />
+          </Transition>
         </template>
-      </p>
-
-      <div class="k-search-container">
-        <KIcon
-          class="k-search-icon"
-          :class="[`theme-${theme}`]"
-          color="currentColor"
-          data-testid="k-code-block-search-icon"
-          icon="search"
-          :size="KUI_ICON_SIZE_40"
-        />
-
-        <label
-          class="k-code-block-search-label"
-          :for="`${id}-search-input`"
-        >
-          <span class="visually-hidden">Search</span>
-        </label>
-
-        <input
-          :id="`${id}-search-input`"
-          ref="codeBlockSearchInput"
-          class="k-code-block-search-input"
-          data-testid="k-code-block-search-input"
-          type="text"
-          @input="handleSearch"
-        >
+      </KInput>
 
-        <p
-          v-if="regExpError !== null"
-          class="k-code-block-search-error"
-        >
-          {{ regExpError.message }}
-        </p>
-
-        <KIcon
-          class="k-is-processing-icon"
-          :class="[`theme-${theme}`, { 'k-is-processing-icon-is-visible': isProcessing }]"
-          color="currentColor"
-          data-testid="k-code-block-is-processing-icon"
-          icon="spinner"
-        />
-
-        <button
-          v-if="query !== ''"
-          appearance="outline"
-          class="k-clear-query-button"
-          data-testid="k-code-block-clear-query-button"
-          title="Clear query"
-          type="button"
-          @click="clearQuery"
+      <Transition name="kongponents-fade-transition">
+        <div
+          v-if="isProcessing || searchQuery"
+          class="code-block-search-results-container"
         >
-          <span class="visually-hidden">Clear query</span>
-
-          <KIcon
-            class="k-clear-icon"
-            :class="[`theme-${theme}`]"
-            color="currentColor"
-            data-testid="k-code-block-clear-icon"
-            icon="clear"
-            :size="KUI_ICON_SIZE_40"
-          />
-        </button>
-      </div>
+          <Transition name="kongponents-fade-transition">
+            <ProgressIcon
+              v-if="isProcessing"
+              class="code-block-processing-icon"
+              :color="getIconColor"
+              data-testid="code-block-processing-icon"
+              :size="KUI_ICON_SIZE_30"
+            />
+          </Transition>
 
-      <div class="k-search-actions">
-        <!-- temporary workaround to get state to "stick" -->
-        <KButton
-          :appearance="isRegExpMode ? 'secondary' : 'tertiary'"
-          :aria-pressed="isRegExpMode"
```

Comment:
**Issue**: This being removed is a regression. The buttons that currently have the `aria-pressed` state are toggle buttons and _need_ this to properly expose their pressed state to the accessibility tree.